client-runtime/src/main/java/com/microsoft/rest/v2/http/HttpClient.java

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.http;
+
+import java.io.IOException;
+
+/**
+ * A generic interface for sending HTTP requests and getting responses.
+ */
+public interface HttpClient {
```

Comment:
@RikkiGibson We need an implementation of this interface that uses RxNetty. We'll have to change the interface to support RxJava, but this is the starting point.